azure-iot-device/azure/iot/device/iothub/sync_clients.py

```diff
@@ -192,8 +193,24 @@ def get_twin(self):
 
         :returns: Twin object which was retrieved from the hub
         """
-        # OPEN QUESTON: does this have a timeout parameter and does it raise on timeout?
-        pass
+        if not self._pipeline.feature_enabled[constant.TWIN]:
+            self._enable_feature(constant.TWIN)
+
+        # hack to work aroud lack of the "nonlocal" keyword in 2.7.  The non-local "context"
+        # object can be read and modified inside the inner function.
+        # (https://stackoverflow.com/a/28433571)
```

Comment:
This is an interesting solution. I've run up against this issue in the past, this is quite clever.